File: frontend/src/renderer/src/stores/team.ts
import { StateCreator } from 'zustand'

export interface TeamConfig {
  id: string
  name: string
  description: string
  strategy: 'synthesize' | 'debate'
  captain: { model?: string; system_prompt_override?: string }
  members: TeamMember[]
}

export type TeamMember =
  | { name: string; source: { persona: string; model?: string; temperature?: number } }
  | { name: string; source: string }

export interface TeamSlice {
  teams: TeamConfig[]
  currentTeamId: string | null
  sessionTeamBindings: Record<string, string>
  setTeams: (teams: TeamConfig[]) => void
  setCurrentTeamId: (id: string | null) => void
  setSessionTeamBinding: (sessionId: string, teamId: string) => void
  getSessionTeam: (sessionId: string) => TeamConfig | undefined
}

export const createTeamSlice: StateCreator<TeamSlice> = (set, get) => ({
  teams: [],
  currentTeamId: null,
  sessionTeamBindings: {},

  setTeams: (teams) => set({ teams }),
  setCurrentTeamId: (currentTeamId) => set({ currentTeamId }),

  setSessionTeamBinding: (sessionId, teamId) => {
    const next = { ...get().sessionTeamBindings }
    if (!teamId) {
      delete next[sessionId]
    } else {
      next[sessionId] = teamId
    }
    set({ sessionTeamBindings: next })
  },

  getSessionTeam: (sessionId) => {
    const state = get()
    const id = state.sessionTeamBindings[sessionId]
    if (id) return state.teams.find((t) => t.id === id)
    return undefined
  },
})

File: backend/crates/loom-types/src/config/team.rs
//! 专家团（Expert Team）配置类型。

use serde::{Deserialize, Serialize};

/// 协作策略
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Default)]
#[serde(rename_all = "snake_case")]
pub enum TeamStrategy {
    /// 合成模式：各成员并行回答 → 团长综合结论
    #[default]
    Synthesize,
    /// 辩论模式：两轮互相质疑后综合结论
    Debate,
}

/// 团长配置（负责汇总/协调）
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct TeamCaptain {
    /// 团长模型，None 表示跟随全局
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub model: Option<String>,
    /// 团长 system prompt 覆写
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub system_prompt_override: Option<String>,
}

/// 成员来源
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum MemberSource {
    /// 内联定义：直接在团队表单中填写配置
    Inline {
        persona: String,
        #[serde(default, skip_serializing_if = "Option::is_none")]
        model: Option<String>,
        #[serde(default, skip_serializing_if = "Option::is_none")]
        temperature: Option<f32>,
    },
    /// 引用已有 Agent 的 config name
    AgentRef(String),
}

/// 团队成员
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TeamMember {
    pub name: String,
    pub source: MemberSource,
}

/// 专家团配置，与 AgentConfig 同级存储。
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TeamConfig {
    /// 唯一标识（UUID v7）
    pub id: String,
    /// 团队名称，如"代码审查团"
    pub name: String,
    /// 用途说明
    #[serde(default)]
    pub description: String,
    /// 协作策略
    #[serde(default)]
    pub strategy: TeamStrategy,
    /// 团长配置
    #[serde(default)]
    pub captain: TeamCaptain,
    /// 成员列表
    #[serde(default)]
    pub members: Vec<TeamMember>,
}
